src/source/Ice/Network.c

```diff
@@ -331,6 +331,43 @@ STATUS socketWrite(INT32 sockfd, const void* pBuffer, SIZE_T length)
     return retStatus;
 }
 
+BOOL isIpAddr(PCHAR hostname)
+{
+    UINT32 ip_1, ip_2, ip_3, ip_4, ip_5, ip_6, ip_7, ip_8;
+    return ((SSCANF(hostname, IPV4_TEMPLATE, &ip_1, &ip_2, &ip_3, &ip_4) == 4 && ip_1 >= 0 && ip_1 <= 255 && ip_2 >= 0 && ip_2 <= 255 && ip_3 >= 0 &&
+             ip_3 <= 255 && ip_4 >= 0 && ip_4 <= 255) ||
+            (SSCANF(hostname, IPV6_TEMPLATE, &ip_1, &ip_2, &ip_3, &ip_4, &ip_5, &ip_6, &ip_7, &ip_8) == 8));
+}
+
+STATUS getIpAddrFromDnsHostname(PCHAR hostname, PCHAR address, UINT16 length)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT8 i = 0, j = 0;
+    UINT16 hostNameLen = length;
+    CHK(hostname != NULL && address != NULL, STATUS_NULL_ARG);
+    // TURN server URLs conform with the public IPv4 DNS hostname format defined here:
+    // https://docs.aws.amazon.com/vpc/latest/userguide/vpc-dns.html#vpc-dns-hostnames
+    // So, we first try to parse the IP from the hostname if it conforms to this format
+    // For example: 35-90-63-38.t-ae7dd61a.kinesisvideo.us-west-2.amazonaws.com
+    // Note: public IPv6 DNS hostnames are not available
+    while (hostNameLen > 0 && hostname[i] != '.') {
```

Comment:
What checks are there to ensure no overrun/overflow occurs for hostname and address?

Reply:
Added check with maxLenDst